### ISSUE TITLE
Classify an Image independently

### ISSUE THREAD

**Issue by rakashi** [suraj-deshmukh/Multi-Label-Image-Classification#5]
Hi, I am trying to do the food image classification, I have collected datasets for 199 classes. I have done classification using **softmax** classifier. But softmax gives sum of all probabilities equal to 1. but I need the independent class probability for each object in image

How do I can achieve this?

**Comment by suraj-deshmukh** [comment]
use **sigmoid** as activation function so that you will get probability for each object in image

**Comment by rakashi** [comment]
I am thinking by using sigmoid with binary_crossentropy is applicable for only two classes but I am having 199 classes.
I have tried 10 classesof data with sigmoid with categorical_crossentropy but I got model with 80% and when I am testing the image its giving all 0's of 10 classes
In clarifai website when I am trying food classification demo that is giving probability independent of classes

**Comment by suraj-deshmukh** [comment]
please go through my repository, i have used **sigmoid** as activation function and **binary_crossentropy** as loss function for multi label image classification having 5 labels(water, tree, mountains, sunset, desert). In your case you have to set 199 neurons in output layer with **sigmoid** as activation function and **binary_crossentropy** as loss function and manually tube the probability threshold values to convert array of probability into 0's and 1's. I hope you will find this piece of information helpful

**Comment by rakashi** [comment]
thanks alot for your response

In your repository, you have converted your dataset into .h5 file. To use your repository I need to convert my dataset into .h5 format. can you please suggest? How can I achieve

**Comment by suraj-deshmukh** [comment]
refer this https://github.com/suraj-deshmukh/Multi-Label-Image-Classification/blob/master/generate-data.py

**Comment by rakashi** [comment]
Yes. I  h seen that, there you have kept miml data.mat file.how did you get that file or we need to convert our dataset into .mat file
can you please suggest?

**Comment by suraj-deshmukh** [comment]
you don't need to convert your data into .mat file. your target variable will be 2d numpy array having shape (no. of examples, no of labels) and another variable having images (shape = (no. of examples, height, width, channel)) and store thos two variables in h5pf file.(refer line 30 to 33 of generate-data.py).
i hope you are getting what i am trying to say
